ruma-events/src/tag.rs

```diff
@@ -50,3 +74,35 @@ impl TagInfo {
         Default::default()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use serde_json::{json, to_value as to_json_value};
+
+    use super::{TagEventContent, TagInfo, TagName, Tags};
+
+    #[test]
+    fn serialization() {
+        let mut tags = Tags::new();
+        tags.insert(TagName::Favourite, TagInfo::new());
+        tags.insert(TagName::LowPriority, TagInfo::new());
+        tags.insert(TagName::ServerNotice, TagInfo::new());
+        tags.insert(TagName::_Custom("u.custom".into()), TagInfo { order: Some(0.9) });
```

Comment:
```suggestion
        tags.insert("u.custom".to_owned().into(), TagInfo { order: Some(0.9) });
```

The tests should be implemented in terms of public API where possible.

Reply:
Also you could use the `maplit`s `btree_map!` macro here.